### PR TITLE
Add more symbolic battery icons

### PR DESCRIPTION
**Pull request by danirabbit**
Towards #257

### CODE REVIEW

File: src/Services/Device.vala [hunk 1]
```diff
@@ -227,7 +227,27 @@ public class Power.Services.Device : Object {
...
+            icon_name += "-%i".printf ((int) (5 * Math.round (percentage / 5)));
+        } else if (percentage >= 10) {
+            icon_name += "10";
+        } else {
+            icon_name += "-0";
+        }
+
+        if (is_charging) {
+            icon_name += "-charging";
+        } else if (time_to_empty == 0 || time_to_empty < 30 * 60) {
```
Comment (danirabbit) [inline src/Services/Device.vala:246]: This feature seems not to be working for color icons?

Comment (danirabbit) [reply src/Services/Device.vala:246]: Honestly it feels kind of aggressive? Maybe it should be like 20 minutes? 15 minutes?

File: src/Services/Device.vala [hunk 2]
```diff
@@ -227,7 +227,27 @@ public class Power.Services.Device : Object {
...
+        if (!is_a_battery) {
+            return "preferences-system-power-symbolic";
+        }
+
+        var icon_name = "battery";
+
+        if (percentage >= 20) {
+            icon_name += "-%i".printf ((int) (5 * Math.round (percentage / 5)));
+        } else if (percentage >= 10) {
+            icon_name += "-10";
```
Comment (danirabbit) [inline src/Services/Device.vala:239]: Maybe this should be < 10. It feels way aggressive to have a single red line at 19% battery

File: src/Services/Device.vala [hunk 3]
```diff
@@ -227,7 +227,27 @@ public class Power.Services.Device : Object {
...
 
     public string get_symbolic_icon_name_for_battery () {
-        return get_icon_name_for_battery () + "-symbolic";
+        if (!is_a_battery) {
+            return "preferences-system-power-symbolic";
+        }
+
+        var icon_name = "battery";
+
+        if (percentage >= 20) {
```
Comment (danirabbit) [inline src/Services/Device.vala:236]: Could be >10 and then add a min range of 20 so we get that caution battery icon until we're under 10%

File: src/Services/Device.vala (hunk 1 above)
Comment (danirabbit) [reply src/Services/Device.vala:246]: This also seems to change the icon if a time estimate is not available, which is not what we want. I think that's why in another place we check that `time_to_emtpy is > 0`. So maybe this needs to be
```suggestion
        } else if (time_to_empty > 0 && time_to_empty < 15 * 60) {
```

File: src/Services/Device.vala [hunk 4]
```diff
@@ -227,7 +227,30 @@ public class Power.Services.Device : Object {
...
+
+        if (percentage > 10) {
+            // Round to the nearest 5 percent
+            // Clamp to 20. There is no 15. Make sure we don't have single px red line until < 10
+            var rounded_percentage = (int) (5 * Math.round (percentage / 5)).clamp (20, 100);
+            icon_name += "-%i".printf (rounded_percentage);
+        } else if (percentage > 0) {
+            icon_name += "-10";
+        } else {
+            icon_name += "-0";
```
Comment (danirabbit) [inline src/Services/Device.vala:244]: should this be `battery-low` and `battery-caution`? Those are the only battery names in [the spec](https://specifications.freedesktop.org/icon-naming-spec/latest/#names), but they are defined as being < 40% and < 20% respectively, which is obviously pretty outdated in terms of what to expect from modern battery life so I highly doubt anyone is respecting those definitions

**Review by micahilbery — APPROVED** [review]
🔋 ❤️ ⚡

**Pull request merged**